src/auth/src/credentials/external_account_sources/executable_sourced.rs

```diff
@@ -0,0 +1,317 @@
+                false,
+                "executables need to be explicitly allowed (set GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES to '1') to run",
+            ));
+        }
+
+        let (command, args) = Self::split_command(command);
+        let output = Command::new(command.clone()).args(&args).output();
+        let output = tokio_timeout(timeout, output.into_future())
+            .await
+            .map_err(|e| CredentialsError::from_source(true, e))?
+            .map_err(|e| CredentialsError::from_source(true, e))?;
+
+        if !output.status.success() {
+            let stderr = String::from_utf8(output.stderr)
+                .map_err(|e| CredentialsError::from_source(false, e))?;
+            let msg = format!("{MSG}, command execution failed, stderr=<{stderr}>");
+            if let Some(code) = output.status.code() {
+                let msg = format!("{msg}, code={code}");
+                return Err(CredentialsError::from_msg(true, msg));
+            };
+            return Err(CredentialsError::from_msg(true, msg));
+        }
+
+        let subject_token = String::from_utf8(output.stdout)
+            .map_err(|e| CredentialsError::from_source(true, e))?
+            .to_string();
+
+        Ok(subject_token)
+    }
+
+    fn split_command(command: String) -> (String, Vec<String>) {
+        let mut parts = command.split_whitespace();
+
+        if let Some(command) = parts.next() {
+            let args: Vec<String> = parts.map(String::from).collect();
+            return (command.to_string(), args);
+        }
+
+        (command, vec![])
+    }
+
+    fn parse_token(output: String) -> Result<String> {
+        let res = serde_json::from_str::<ExecutableResponse>(output.as_str())
+            .map_err(|e| CredentialsError::from_source(false, e))?;
+
+        if !res.success {
+            return Err(res.to_cred_error());
+        }
+
+        match res.token_type.as_str() {
+            JWT_TOKEN_TYPE | ACCESS_TOKEN_TYPE => match res.id_token {
+                Some(id_token) => Ok(id_token),
+                None => Err(CredentialsError::from_msg(
+                    false,
+                    "missing `id_token` field",
+                )),
+            },
+            SAML2_TOKEN_TYPE => match res.saml_response {
+                Some(saml_response) => Ok(saml_response),
+                None => Err(CredentialsError::from_msg(
+                    false,
+                    "missing `saml_response` field",
+                )),
+            },
+            _ => Err(CredentialsError::from_msg(
+                false,
+                "contains unsupported token type",
+            )),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::constants::JWT_TOKEN_TYPE;
+    use scoped_env::ScopedEnv;
+    use serde_json::json;
+    use serial_test::serial;
+    use tokio::time::{Duration, Instant};
+
+    type TestResult = std::result::Result<(), Box<dyn std::error::Error>>;
```

Comment:
Consider: `type TestResult = anyhow::Result<()>` we started using that everywhere else.